### PR TITLE
add PhotoRail class

### PR DESCRIPTION
**Pull request by maxhbr**

### CODE REVIEW

**Comment by maxhbr** [comment]
This PR is still work in progress

File: src/main.cpp.orig [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#include <Arduino.h>
```
Comment (hackenbergstefan) [inline src/main.cpp.orig]: Forgot to delete?

File: src/PhotoRail.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+#include "Rail.h"
+
+class PhotoRail
+ : public Rail
+{
+private:
+    int left = 0;
+    int right = 0;
+
+    IrSony ir_sony;
```
Comment (hackenbergstefan) [inline src/PhotoRail.h:16]: We should abstract this to interface; sth like:
`class CameraRemote { virtual void shoot(); }`

File: src/main.cpp.orig (hunk 1 above)
Comment (maxhbr) [reply src/main.cpp.orig]: sorry, shouldn't be added. missing in `.gitignore`, will fix it.

File: src/PhotoRail.h (hunk 2 above)
Comment (maxhbr) [reply src/PhotoRail.h:16]: sounds good! I will try it.

Comment (maxhbr) [reply src/PhotoRail.h:16]: or better: let us do it after the merge

**Review by hackenbergstefan — APPROVED** [review]

**Pull request merged**